### PR TITLE
Micro optimizations

### PR DESCRIPTION
**Pull request by ErrorNoInternet**

### CODE REVIEW

File: src/desktop.rs [hunk 1]
```diff
@@ -1,21 +1,13 @@
...
-
     // Retrieve the environment variables and handle Result types
     let desktop_env = std::env::var("XDG_CURRENT_DESKTOP");
     let display_backend_result = std::env::var("XDG_SESSION_TYPE");
 
     // Capitalize the first letter of the display backend value
-    let display_backend = capitalize_first_letter(display_backend_result.as_deref().unwrap_or(""));
+    let mut display_backend = display_backend_result.unwrap_or_default();
+    if let Some(c) = display_backend.as_mut_str().get_mut(0..1) {
+        c.make_ascii_uppercase();
```
Comment (NotAShelf) [inline src/desktop.rs:9]: I didn't know about this, cool

File: src/main.rs [hunk 2]
```diff
@@ -71,5 +74,6 @@ fn print_system_info(fields: &Fields) {
  {BLUE}   ▟█▛{CYAN}▗█▖       {CYAN}▟█▛          {CYAN}  {BLUE}Desktop{RESET}       {desktop}
  {BLUE}  ▝█▛  {CYAN}██▖{BLUE}▗▄▄▄▄▄▄▄▄▄▄▄       {CYAN}󰍛  {BLUE}Memory{RESET}        {memory_usage}
  {BLUE}   ▝  {CYAN}▟█▜█▖{BLUE}▀▀▀▀▀██▛▀▀▘       {CYAN}󱥎  {BLUE}Storage (/){RESET}   {storage}
- {CYAN}     ▟█▘ ▜█▖    {BLUE}▝█▛          {CYAN}  {BLUE}Colors{RESET}        {colors}");
+ {CYAN}     ▟█▘ ▜█▖    {BLUE}▝█▛          {CYAN}  {BLUE}Colors{RESET}        {colors}
+").as_bytes());
```
Comment (NotAShelf) [inline src/main.rs:78]: what does this do?

**Review by NotAShelf — COMMENTED** [review]
lgtm, no notes

just a little confused about the addition of as_byes

File: src/main.rs (hunk 2 above)
Comment (ErrorNoInternet) [reply src/main.rs:78]: `std::io::stdout().write_all()` takes a `&[u8]`

Comment (NotAShelf) [reply src/main.rs:78]: Okay I guess I should've asked about the whole change. Why write_all() and friends?

Comment (ErrorNoInternet) [reply src/main.rs:78]: using println!() makes multiple write syscalls, one for each line (can see from strace), whereas write_all writes the whole thing (multiple lines) in one go with a single write syscall

**Pull request merged**